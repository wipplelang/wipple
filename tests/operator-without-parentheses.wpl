show 1 + 2
show True or False
show (1 + "hi")
