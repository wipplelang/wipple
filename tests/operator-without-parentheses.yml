code: |
    show 1 + 2
    show True or False
    show (1 + "hi")

output: |

diagnostics: |
    error: cannot add `()` to `Number`
       ┌─ test:1:8
       │
     1 │ show 1 + 2
       │        ^
       │        │
       │        could not find instance `Add () Number _`
       │        operators consume all expressions on either side; you may be missing parentheses
       │
       ┌─ <dir>/std/math.wpl:16:14
       │
    16 │   'a + 'b -> Add 'a 'b
       │              --- actual error occurred here
       │
       = for more information, see https://wipple.dev/playground/?lesson=errors/missing-instance

    error: missing instance
       ┌─ test:2:11
       │
     2 │ show True or False
       │           ^^
       │           │
       │           could not find instance `Or () Boolean _`
       │           operators consume all expressions on either side; you may be missing parentheses
       │
       ┌─ <dir>/std/logic.wpl:57:15
       │
    57 │   'a or 'b -> Or 'a (() -> 'b)
       │               -- actual error occurred here
       │
       = for more information, see https://wipple.dev/playground/?lesson=errors/missing-instance

    error: cannot add `Number` to `Text`
       ┌─ test:3:9
       │
     3 │ show (1 + "hi")
       │         ^ could not find instance `Add Number Text _`
       │
       ┌─ <dir>/std/math.wpl:16:14
       │
    16 │   'a + 'b -> Add 'a 'b
       │              --- actual error occurred here
       │
       = for more information, see https://wipple.dev/playground/?lesson=errors/missing-instance
