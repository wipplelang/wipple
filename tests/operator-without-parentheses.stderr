error: cannot add `show 1` to `2`
   ┌─ test:1:8
   │
 1 │ show 1 + 2
   │ ------ ^ - this has type `Number`
   │ │      │  
   │ │      could not find instance `Add () Number _`
   │ │      `+` consumes all expressions on either side; you may be missing parentheses
   │ this has type `()`
   │ this is parsed as one single input to `+`
   │
   ┌─ <dir>/std/math.wpl:17:14
   │
17 │   'a + 'b -> Add 'b 'a
   │              --- actual error occurred here
   │
   = add parentheses around `1 + 2`: `(1 + 2)`
   = for more information, see https://wipple.dev/playground/?lesson=errors/missing-instance

error: missing instance
   ┌─ test:2:11
   │
 2 │ show True or False
   │ --------- ^^
   │ │         │
   │ │         could not find instance `Or () Boolean _`
   │ │         `or` consumes all expressions on either side; you may be missing parentheses
   │ this has type `()`
   │ this is parsed as one single input to `or`
   │
   ┌─ <dir>/std/logic.wpl:54:15
   │
54 │   'a or 'b -> Or (() -> 'b) 'a
   │               -- actual error occurred here
   │
   = add parentheses around `True or False`: `(True or False)`
   = for more information, see https://wipple.dev/playground/?lesson=errors/missing-instance

error: cannot add `1` to `"hi"`
   ┌─ test:3:9
   │
 3 │ show (1 + "hi")
   │       - ^ ---- this has type `Text`
   │       │ │  
   │       │ could not find instance `Add Number Text _`
   │       this has type `Number`
   │
   ┌─ <dir>/std/math.wpl:17:14
   │
17 │   'a + 'b -> Add 'b 'a
   │              --- actual error occurred here
   │
   = for more information, see https://wipple.dev/playground/?lesson=errors/missing-instance

